modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBaseReindexRestHandler.java

```diff
@@ -61,7 +76,11 @@ protected AbstractBaseReindexRestHandler(Settings settings, Client client,
     }
 
     protected void execute(RestRequest request, Request internalRequest, RestChannel channel) throws IOException {
-        internalRequest.setRequestsPerSecond(request.paramAsFloat("requests_per_second", internalRequest.getRequestsPerSecond()));
+        Float requestsPerSecond = parseRequestsPerSecond(request);
+        if (requestsPerSecond != null) {
```

Comment:
I guess I like it better this way because null doesn't make sense in the context of the other caller. Does that make sense?